core/src/transport.rs

```diff
@@ -245,7 +252,7 @@ pub struct ListenerId(u64);
 impl ListenerId {
     /// Creates a new `ListenerId`.
     pub fn new() -> Self {
```

Comment:
To be consistent with `ConnectionId`, let's deprecate this one and add a new `ConnectionId::next` function.

Reply:
To avoid more confusion, I think it also makes sense to remove the `Default` implementation. That will be a breaking change but that is okay. To make the transition easier, let's:

- Remove the `Default` implementation
- Add a `pub fn default()` function
- Deprecate that function and refer to `next`

This will still break users if they rely on the `Default` trait bound but I'd expect that to be uncommon.

Reply:
All done. Also, I added allowed `clippy::new_without_default` and `clippy::should_implement_trait` to the new and default functions, though those will be removed after a couple of releases. Wouldnt know if CI would complain here about them so I added them just in case.